scipy/stats/_qmc.py

```diff
@@ -1223,6 +1223,8 @@ class Sobol(QMCEngine):
     bits : int, optional
         Number of bits of the generator. Control the maximum number of points
         that can be generated, which is ``2**bits``. Maximal value is 64.
+        It also conditions the return type to be either 32 or 64 bits arrays.
+        Default is None and use 30 bits for backward compatibility.
```

Comment:
Although with the above, this suggests that it will return a 32-bit array, which I don't think is right.

Reply:
Yes, I will add something. Shall I constrain the dtype to be 64bits for backward compatibility or do you think it's ok to have a dtype change without notice. As in this case we don't lose precision.

Reply:
> As in this case we don't lose precision.
The very first points of sequences are different.

```python3
rng = Sobol(1)
x = rng.random()
y = x.astype(np.float32)
x-y  # array([[1.86264515e-08]])
```
Unrelated: I just tried `rng = Sobol(1); x = rng.random(100)` and got the expected warning, but it needs `stacklevel`._

Reply:
Ah good catch!

Ok I will send an email.